herd/AArch64Sem.ml

```diff
@@ -2093,13 +2096,31 @@ module Make
             lift_memop rn Dir.W true memtag mop_fail_with_wb (to_perms "rw" sz)
               (read_reg_ord rn ii) (read_reg_data sz rt ii) an ii
           )(
-            (* CAS does not generate an Explicit Write Effect          *)
-            (* It is IMPLEMENTATION SPECIFIC if there is an update to  *)
-            (*                                the dirty bit of the TTD *)
-            (* Note: the combination of dir=Dir.R and updatedb=true    *)
-            (*                    triggers an alternative in check_ptw *)
-            lift_memop rn Dir.R true memtag mop_fail_no_wb (to_perms "rw" sz)
-              (read_reg_ord rn ii) (read_reg_data sz rt ii) an ii
+              (* CAS does not generate an Explicit Write Effect          *)
+              (* It is IMPLEMENTATION SPECIFIC if there is an update to  *)
+              (*                                the dirty bit of the TTD *)
+              (* Note: the combination of dir=Dir.R and updatedb=true    *)
+              (*                    triggers an alternative in check_ptw *)
+              let ma = read_reg_ord rn ii
+              and action updatedb checked ma =
+                lift_memop rn Dir.R updatedb checked mop_fail_no_wb (to_perms "rw" sz)
+                  ma (read_reg_data sz rt ii) an ii in
+               if memtag && C.mte_store_only then
+                M.altT (
```

Comment:
This is just checking if I understand the code correctly: would it consider two options when there is no Explicit Write Effect:
* no db bit update, no tagcheck
* with the db bit update, with the tagcheck

? If yes, do we already know if this actually needs to be four cases (yes/no to each part)?